Extension/.scripts/test.ts

```diff
@@ -0,0 +1,229 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { runTests } from '@vscode/test-electron';
+import { spawnSync } from 'child_process';
+import { CommentArray, CommentObject } from 'comment-json';
+import { readdir } from 'fs/promises';
+import { resolve } from 'path';
+import { env } from 'process';
+import { returns } from '../src/Utility/Async/returns';
+import { filepath } from '../src/Utility/Filesystem/filepath';
+import { is } from '../src/Utility/System/guards';
+import { verbose } from '../src/Utility/Text/streams';
+import { getTestInfo } from '../test/common/selectTests';
+import { $args, $root, $scenario, brightGreen, checkFile, checkFolder, cmdSwitch, cyan, error, gray, green, readJson, red, writeJson } from './common';
```

Comment:
What is the significance of the variables that start with $?

Reply:
It's just a batch scripting convention -- when building 'scripts' for project infrastructure I just fall into that pattern. 
And what's life without whimsey 😄